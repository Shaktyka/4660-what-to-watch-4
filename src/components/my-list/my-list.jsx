import React from 'react';

import PropTypes from 'prop-types';
import PageHeader from '../page-header/page-header.jsx';
import UserBlock from '../user-block/user-block.jsx';
import PageFooter from '../page-footer/page-footer.jsx';

const getPageTitle = () => {
  return (
    <h1 className="page-title user-page__title">My list</h1>
  );
};

const MyList = (props) => {
  const {userData} = props;

  return (
    <div className="user-page">
      <PageHeader uniqueClass={`user-page__head`}>
        {getPageTitle()}
        <UserBlock
          isAuthorized={true}
          userData={userData}
        />
      </PageHeader>

      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>

        <div className="catalog__movies-list">

          <article className="small-movie-card catalog__movies-card">
            <div className="small-movie-card__image">
              <img src="img/what-we-do-in-the-shadows.jpg" alt="What We Do in the Shadows" width="280" height="175" />
            </div>
            <h3 className="small-movie-card__title">
              <a className="small-movie-card__link" href="movie-page.html">What We Do in the Shadows</a>
            </h3>
          </article>

        </div>
      </section>

      <PageFooter />
    </div>
  );
};

MyList.propTypes = {
  userData: PropTypes.object
};

export default MyList;
