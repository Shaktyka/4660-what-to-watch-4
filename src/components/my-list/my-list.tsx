import * as React from 'react';

import {connect} from 'react-redux';
import {getUserData} from '../../reducer/user/selectors';
import {Operation as DataOperation} from '../../reducer/data/data';
import {
  getFavoritesFilms,
  getIsFavoritesFilmsLoading,
  getLoadFavoritesFilmsError
} from '../../reducer/data/selectors';

import PageHeader from '../page-header/page-header';
import UserBlock from '../user-block/user-block';
import PageFooter from '../page-footer/page-footer';
import Loader from '../loader/loader';

import withActiveItem from '../../hocs/with-active-item/with-active-item';
import withShowMore from '../../hocs/with-show-more/with-show-more';
import MoviesList from '../movies-list/movies-list';
import {FilmInterface, UserDataInterface} from '../../types.ts';

const MoviesListWrapped = withActiveItem(withShowMore(MoviesList));

const getPageTitle = () => {
  return (
    <h1 className="page-title user-page__title">My list</h1>
  );
};

interface MyListProps {
  userData: UserDataInterface;
  favoritesFilms: Array<FilmInterface>;
  isFavoritesFilmsLoading: boolean;
  loadFavoritesFilmsError: string;
}

class MyList extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const {loadFavoriteFilms} = this.props;
    loadFavoriteFilms();
  }

  render() {
    const {
      userData,
      favoritesFilms,
      isFavoritesFilmsLoading,
      loadFavoritesFilmsError
    } = this.props;

    return (
      <div className="user-page">
        <PageHeader uniqueClass={`user-page__head`}>
          {getPageTitle()}
          <UserBlock
            isAuthorized={true}
            userData={userData}
          />
        </PageHeader>

        <section className="catalog">
          <h2 className="catalog__title visually-hidden">Catalog</h2>

          {
            isFavoritesFilmsLoading
              ?
              <Loader />
              :
              <MoviesListWrapped
                films={favoritesFilms}
                isLoading={isFavoritesFilmsLoading}
                loadFilmsError={loadFavoritesFilmsError}
              />
          }

        </section>

        <PageFooter />
      </div>
    );
  }
}


const mapStateToProps = (state) => ({
  userData: getUserData(state),
  favoritesFilms: getFavoritesFilms(state),
  isFavoritesFilmsLoading: getIsFavoritesFilmsLoading(state),
  loadFavoritesFilmsError: getLoadFavoritesFilmsError(state)
});

const mapDispatchToProps = (dispatch) => ({
  loadFavoriteFilms() {
    dispatch(DataOperation.loadFavoriteFilms());
  }
});

export {MyList};
export default connect(mapStateToProps, mapDispatchToProps)(MyList);
