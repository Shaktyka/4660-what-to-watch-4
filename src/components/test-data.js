export const noop = () => {
  // Mock function for test props
};

export const MOVIE_NAV_TABS = [`Overview`, `Details`, `Reviews`];

export const GENRES = [`All genres`, `Crime`, `Sci-Fi`, `Drama`, `Comedy`];

export const filmData = {
  bgColor: `#A6B7AC`,
  cover: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
  description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City
    seeking revenge against Bill the Butcher, his father's killer.`,
  director: `Martin Scorsese`,
  duration: 167,
  genre: `Crime`,
  id: 1,
  isFavorite: false,
  poster: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
  preview: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  ratingCount: 370881,
  ratingScore: 8.8,
  source: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  starring: [`Leonardo DiCaprio`, `Cameron Diaz`, `Daniel Day-Lewis`],
  title: `Gangs of new york`,
  year: 2002
};

export const userData = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatar: `img/1.png`
};

export const comment = {
  id: 1,
  userId: 4,
  userName: `Kate Muir`,
  rating: 8.9,
  comment: `Discerning travellers`,
  date: `2019-05-08T14:13:56.569Z`
};

export const REVIEWS = [
  {
    id: 1,
    authorId: 4,
    authorName: `Kate Muir`,
    rating: 8.9,
    text: `Discerning travellers`,
    date: `2019-05-08T14:13:56.569Z`
  }
];
