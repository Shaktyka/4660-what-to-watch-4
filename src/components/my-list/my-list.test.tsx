import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {BrowserRouter} from 'react-router-dom';
import {userData, filmData, noop} from '../test-data';

import {MyList} from './my-list';

describe(`MyList rendering`, () => {

  it(`MyList renders correctly`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <MyList
              userData={userData}
              favoritesFilms={[filmData]}
              isFavoritesFilmsLoading={false}
              loadFavoritesFilmsError={``}
              loadFavoriteFilms={noop}
            />
          </BrowserRouter>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
