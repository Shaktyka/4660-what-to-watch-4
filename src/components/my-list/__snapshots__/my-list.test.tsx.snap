// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyList rendering MyList renders correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        className="user-block__avatar"
        href="/mylist"
        onClick={[Function]}
        style={
          Object {
            "display": "block",
          }
        }
      >
        <img
          alt="Oliver.conner's avatar"
          height="63"
          src="https://4.react.pages.academyimg/1.png"
          width="63"
        />
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <a
        className="small-movie-card catalog__movies-card"
        href="/films/1"
        onClick={[Function]}
      >
        <article
          id={1}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="175"
              poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg"
              type="video/webm"
              width="280"
            >
              Sorry, your browser doesn't support embedded video
            </video>
          </div>
          <h3
            className="small-movie-card__title"
            style={
              Object {
                "color": "#d9ca74",
              }
            }
          >
            Gangs of new york
          </h3>
        </article>
      </a>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
